code/include/playfab/ClientLoginApi.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <playfab/PlayFabClientDataModels.h>
+#include <playfab/PlayFabError.h>
+#include <playfab/TaskQueue.h>
+
+namespace PlayFab
+{
+    class CallRequestContainerBase;
+    class CallRequestContainer;
+    class PlayFabApiSettings;
+
+    class ClientLoginApi
+    {
+    public:
+        void LoginWithCustomID(const PlayFabLoginWithCustomIDRequest& request, const ProcessApiCallback<ClientModels::LoginResult> callback, const TaskQueue& taskQueue, const ErrorCallback errorCallback = nullptr, void* customData = nullptr);
```

Comment:
out of curiosity, what is the calling pattern here?  do you need to make a "ClientLoginApi" instance and call the instance method?

Reply:
This is not a public class.  A ClientLoginApi instance will be a member of the PlayFabGlobalState object returned from PlayFabInitialize.  Public login methods will take a PlayFabStateHandle and that instance will be used to login.

---
In reply to: [595554331](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r595554331) [](ancestors = 595554331)

Reply:
@jplafonta this class is provided in a header inside the include directory. 
Additionally, the class itself doesn't have any restrictions which prevent callers from instantiating one.

For all intents and purposes this says "public class" to me. Maybe the privateness would be more clear if this was moved under `code/source`

Reply:
Ya forget about the source layout now.  Assume the public headers for the C sdk will eventually only be those in code/include/playfab_c.

---
In reply to: [596215506](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r596215506) [](ancestors = 596215506)

Reply:
it'd be helpful if as part of this review you moved around the necessary files to reflect a layout which suggests what is or isn't public.
otherwise it's not clear what pieces I should be thinking of as public or not.

---
In reply to: [596221029](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r596221029) [](ancestors = 596221029,596215506)

Reply:
oh. maybe the answer is that the only files part of the public interface are the files in code/include/playfab_c ?

---
In reply to: [596372153](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r596372153) [](ancestors = 596372153,596221029,596215506)

Reply:
oh doh! I just realized you made a note of these things in your original email. apologies for missing that. I only checked this PR description and didn't see any such hints (maybe valuable to add in retrospect)

Reply:
Ya that's correct.  Sorry for the confusion

---
In reply to: [596375889](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r596375889) [](ancestors = 596375889,596372153,596221029,596215506)

Reply:
Good idea...updated the github PR with a little more context. 

---
In reply to: [596394945](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r596394945) [](ancestors = 596394945)